pkg/appmgmt/appmgmt_recovery_test.go

```diff
@@ -197,6 +197,40 @@ func TestAppStatesDuringRecovery(t *testing.T) {
 	assert.Equal(t, app02.GetApplicationState(), cache.ApplicationStates().Accepted)
 }
 
+func TestPodRecovery(t *testing.T) {
+	conf.GetSchedulerConf().OperatorPlugins = "mocked-app-manager"
+	amProtocol := cache.NewMockedAMProtocol()
+	apiProvider := client.NewMockedAPIProvider(false)
+	taskRequests := make([]*interfaces.AddTaskRequest, 0)
+	amProtocol.UseAddTaskFn(func(request *interfaces.AddTaskRequest) {
+		taskRequests = append(taskRequests, request)
+	})
+	amService := NewAMService(amProtocol, apiProvider)
+	amService.register(&mockedAppManager{})
+
+	apps, err := amService.recoverApps()
+	assert.NilError(t, err)
+	assert.Equal(t, 4, len(taskRequests))
+	assert.Equal(t, 2, len(apps))
+
+	expected := map[string]map[string]bool{
+		"app01": {
+			"task01": true,
+			"task02": true,
+		},
+		"app02": {
+			"task04": true,
+			"task05": true,
```

Comment:
I haven't deeply analized what happens here, but don't we want some variety? In all cases, we expect true, so why do we have this here?